framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -0,0 +1,635 @@
+     * @param printWriter the writer where the class definition should be printed
+     * @return the number of outer classes within which this class is nested
+     */
+    private static int printClassDefinitions(
+            String basename, AClassWrapper aClass, PrintWriter printWriter) {
+
+        String nameToPrint = basename;
+        String remainingInnerClassNames = "";
+        if (basename.contains("$")) {
+            nameToPrint = basename.substring(0, basename.indexOf('$'));
+            remainingInnerClassNames = basename.substring(basename.indexOf('$') + 1);
+        }
+
+        // For any outer class, print "class".  For a leaf class, print "enum" or "class".
+        if ("".equals(remainingInnerClassNames) && aClass.isEnum()) {
+            printWriter.print("enum ");
+        } else {
+            printWriter.print("class ");
+        }
+        printWriter.print(formatAnnotations(aClass.getAnnotations()));
+        printWriter.print(nameToPrint);
+        printTypeParameters(aClass, printWriter);
+        printWriter.println(" {");
+        printWriter.println();
+        if ("".equals(remainingInnerClassNames)) {
+            return 1; // once no inner classes are left,
+            // one curly brace is still required for the original class
+        } else {
+            return 1 + printClassDefinitions(remainingInnerClassNames, aClass, printWriter);
+        }
+    }
+
+    /**
+     * Prints all the fields of a given class
+     *
+     * @param aClass the class whose fields should be printed
+     * @param printWriter the writer on which to print the fields
+     */
+    private static void printFields(AClassWrapper aClass, PrintWriter printWriter) {
+
+        if (aClass.getFields().keySet().size() != 0) {
+            printWriter.println(INDENT + "// fields:");
+            printWriter.println();
+        }
+
+        for (Map.Entry<String, AFieldWrapper> fieldEntry : aClass.getFields().entrySet()) {
+            String fieldName = fieldEntry.getKey();
+            AFieldWrapper aField = fieldEntry.getValue();
+            printField(aField, fieldName, printWriter);
+        }
+    }
+
+    /**
+     * Prints a field declaration, including a trailing semi-colon and a newline.
+     *
+     * @param aField the field declaration
+     * @param fieldName the name of the field
+     * @param printWriter the writer on which to print
+     */
+    private static void printField(
+            AFieldWrapper aField, String fieldName, PrintWriter printWriter) {
+        printWriter.print(INDENT);
+        printWriter.print(formatAFieldImpl(aField, fieldName, null));
+        printWriter.println(";");
+        printWriter.println();
+    }
+
+    /**
+     * Prints a method signature in stub file format (i.e., without a method body).
+     *
+     * @param aMethodWrapper the method to print
+     * @param basename the simple name of the containing class. Used only to determine if the method
```

Comment:
Line 469 below checks for "." in `basename`.  However, a simple name never contains ".".  I suspect this documentation is incorrect regarding the format of the argument.